mlir/include/mlir/Dialect/MIOpen/Tuning/GridwiseGemmParams.h

```diff
@@ -366,12 +366,12 @@ class PopulateParamsBase {
     // in the algorithm.
     int64_t vectorizationSize = 1;
     auto dataType = ctx.getDataType();
-    if (dataType.isF32()) {
-      vectorizationSize = 4;
-    } else if (dataType.isF16() || dataType.isBF16()) {
-      // FIXME: figure out the best vectorization length for f16 and bf16.
-      vectorizationSize = 4;
-    }
+    unsigned dataWidth = dataType.getIntOrFloatBitWidth();
+    // 128 is the upper limit we support in vectorized load/store, which could
+    // be 4 fp32, 8 fp16, or 16 int8
+    const auto highestPotentialVectorizationLen = 128;
```

Comment:
Can this be an actual concrete datatype, maybe `const size_t`?

Heck, let's make it `constexpr` while we're here

Reply:
I don't see much value in addition to const in this context so prefer to just use const instead constexpr. I'd be declaring a constexpr if we use an expression to derive the value of `highestPotentialVectorizationLen`, and that we know for sure this can be done at compile time.